src/Voron/Data/Lookups/Lookup.Iterator.cs

```diff
@@ -276,8 +295,11 @@ public void Seek<T>(T key)
                 _tree.FindPageFor(ref lookupKey, ref _cursor);
 
                 ref var state = ref _cursor._stk[_cursor._pos];
-                if (state.LastSearchPosition < 0)
-                    state.LastSearchPosition = Math.Min(~state.LastSearchPosition, state.Header->NumberOfEntries - 1);
+                
+                if (state.LastSearchPosition >= 0)
+                    return;
+
+                state.LastSearchPosition = ~state.LastSearchPosition - 1;
```

Comment:
What if this is `~LastSearchPosition` is 0? Then then the `LastSearchPosition` would be -1, no?

Reply:
Negative `LastSearchPosition` (set in `SearchInCurrentPage`) means we didn't find the seeked key. In such case we point to the biggest value smaller than seeked key, then move `LastSearchPosition` one position forward, effectively pointing to the smallest value bigger than seeked key. This behavior works for `ForwardIterator`, but for `BackwardIterator` we have to go back one position.

We get negative value here when seeking key that's smaller than any value in the tree. It's then handled by `MoveNext` that attempts to go to previous page, but since there's no more pages it returns `false`.